### DISCUSSION TITLE
Linter to rewrite `err=foo; if err` to `if err=foo; err`

### DISCUSSION THREAD

**Discussion by grosser** [golangci/golangci-lint#4755] [Linter Ideas]
### Your feature request related to a problem? Please describe.

rewrite
```
err = foo
if err != nil {
```

into
```
if err = foo; err != nil {
```

so our code looks consistent

### Describe the solution you'd like.

automatic reformat

### Describe alternatives you've considered.

manually reformatting

**Comment by ldez** [comment]
Hello,

FYI golangci-lint is a wrapper around linters.

Also, I think you are looking for something like [`ifshort`](https://github.com/esimonov/ifshort) or [`errifinline`](https://github.com/bastianccm/errifinline).

[ifshort](https://github.com/esimonov/ifshort) has been archived.

[`errifinline`](https://github.com/bastianccm/errifinline) is will maybe be transformed https://github.com/golangci/golangci-lint/pull/4385#issuecomment-1946111767

**Comment by grosser** [comment]
awesome, thx!

**Comment by grosser** [comment]
can I use this without the [PR](https://github.com/golangci/golangci-lint/pull/4385) being merged ?
... don't really want to setup a whole new flow for this 1 linter

**Comment by bombsimon** [comment]
Not unless you build `golangci-lint` from the branch of the PR, but then you'll miss other things instead unless you rebase manually.
